src/test/java/com/github/checkstyle/regression/configuration/ConfigGeneratorTest.java

```diff
@@ -0,0 +1,164 @@
+
+        final File actual = generateConfig(Collections.emptyList());
+        assertTrue("Config is not as expected", FileUtils.contentEquals(excepted, actual));
+    }
+
+    @Test
+    public void testGenerateConfigTextWithCheckerParentModule() throws Exception {
+        final File excepted = getExpectedXml("expected_checker_parent_module.xml");
+
+        final ModuleExtractInfo extractInfo = ImmutableModuleExtractInfo.builder()
+                .name("FileLengthCheck")
+                .packageName(BASE_PACKAGE + ".checks.sizes")
+                .parent("Checker")
+                .build();
+        final ModuleInfo moduleInfo = ImmutableModuleInfo.builder()
+                .moduleExtractInfo(extractInfo)
+                .build();
+
+        final File actual = generateConfig(Collections.singletonList(moduleInfo));
+
+        assertTrue("Config is not as expected", FileUtils.contentEquals(excepted, actual));
+    }
+
+    @Test
+    public void testGenerateConfigTextWithTreeWalkerParentModule() throws Exception {
+        final File excepted = getExpectedXml("expected_tree_walker_parent_module.xml");
+
+        final ModuleExtractInfo extractInfo = ImmutableModuleExtractInfo.builder()
+                .name("HiddenFieldCheck")
+                .packageName(BASE_PACKAGE + ".checks.coding")
+                .parent("TreeWalker")
+                .build();
+        final ModuleInfo moduleInfo = ImmutableModuleInfo.builder()
+                .moduleExtractInfo(extractInfo)
+                .build();
+
+        final File actual = generateConfig(Collections.singletonList(moduleInfo));
+        assertTrue("Config is not as expected", FileUtils.contentEquals(excepted, actual));
+    }
+
+    @Test
+    public void testGenerateConfigTextWithMiscModuleInfos() throws Exception {
+        final File excepted = getExpectedXml("expected_misc_module_infos.xml");
+
+        final ModuleExtractInfo extractInfo1 = ImmutableModuleExtractInfo.builder()
+                .name("NewlineAtEndOfFileCheck")
+                .packageName(BASE_PACKAGE + ".checks")
+                .parent("Checker")
+                .build();
+        final ModuleInfo moduleInfo1 = ImmutableModuleInfo.builder()
+                .moduleExtractInfo(extractInfo1)
+                .build();
+
+        final ModuleExtractInfo extractInfo2 = ImmutableModuleExtractInfo.builder()
+                .name("EmptyStatementCheck")
+                .packageName(BASE_PACKAGE + ".checks.coding")
+                .parent("TreeWalker")
+                .build();
+        final ModuleInfo moduleInfo2 = ImmutableModuleInfo.builder()
+                .moduleExtractInfo(extractInfo2)
+                .build();
+
+        final File actual = generateConfig(Arrays.asList(moduleInfo1, moduleInfo2));
+        assertTrue("Config is not as expected", FileUtils.contentEquals(excepted, actual));
+    }
+
+    @After
+    public void tearDown() throws Exception {
```

Comment:
`DiffParserTest` had this at the top of the class while in here you put in the middle.
Let's be consistent and always have `After`/`Before` at the top of the class so it is the first thing users see when reading this file.